myapp/utility.py

```diff
@@ -1,116 +1,880 @@
+def serialize_class_name_list():
+    return [c.name for c in query_all_class()]
+
+
+# return a list of professors' names
+def serialize_prof_name_list():
+    return [u.name for u in query_all_admin()] + [u.name for u in query_all_superadmin()]
+
+
+# convert a query for admins into a list of JSON objects
+def serialize_admin_list(admins_query):
+    current_admins = []
+    for admin in admins_query:
+        classes = [c.id for c in admin.classes]
+        labs = [lab.id for lab in admin.labs]
+        current_admins.append({'id': admin.id,
+                               'username': admin.username,
+                               'password': admin.password,
+                               'name': admin.name, 'classes': classes,
+                               'labs': labs})
+    return current_admins
+
+
+# convert a query for classes into a list of JSON objects
+def serialize_class_list(classes_query):
+    current_classes = []
+    for c in classes_query:
+        professors = [p.username for p in c.users if (p.role_name == 'Admin' or p.role_name == 'SuperAdmin')]
+        students = [s.username for s in c.users if s.role_name == 'Student']
+        labs = [lab.id for lab in c.labs]
+        current_classes.append({'id': c.id,
+                                'name': c.name,
+                                'time': c.time,
+                                'professors': professors,
+                                'students': students,
+                                'labs': labs})
+    return current_classes
+
+
+# check the data sent by client and
+# deserialize the information of a lab from jsonData sent by client
+# and serialize it into JSON format dictionary that is consistent with
+# python side naming convention
+def deserialize_lab(lab_json):
+    err_msg = check_existence(lab_json, 'labName', 'className', 'classTime',
+                                        'professorName', 'labDescription',
+                                        'experiments')
+    if err_msg != '':
+        return (dict(), err_msg)
+
+    lab_info = {'name': lab_json['labName'],
+                'class_name': lab_json['className'],
+                'class_time': lab_json['classTime'],
+                'prof_name': lab_json['professorName'],
+                'description': lab_json['labDescription'],
+                'experiments': lab_json['experiments'],
+                'old_lab_id': lab_json['oldLabId']}
+
+    for e in lab_json['experiments']:
+        err_msg = check_existence(e, 'name', 'description', 'order',
+                                     'valueType', 'valueRange',
+                                     'valueCandidates')
+        if err_msg != '':
+            return (lab_info, err_msg)
+    return (lab_info, '')
+
+
+# Covert a string in the format '[a,b,c]' into a python list
+def deserialize_lab_id(lab_ids):
+    lab_ids = lab_ids.lstrip('\"[').rstrip('\"]').split(',')
+    lab_ids = [lab.lstrip('\'').rstrip('\'') for lab in lab_ids if
```

Comment:
despite the comment, the format appears to be "['a','b','c']" which is very nearly valid json. Why not just use a json array?